### ISSUE TITLE
Implement `node_register_module` and some `V8::` APIs

### ISSUE THREAD

**Issue by Jarred-Sumner** [oven-sh/bun#3110]
This is blocking:
- [`node-canvas`](https://github.com/Automattic/node-canvas)
- ~[`bcrypt`](https://www.npmjs.com/package/bcrypt?activeTab=readme)~
- [`better-sqlite3`](https://www.npmjs.com/package/better-sqlite3)

These (and others like uWebSocket.js) use the V8 APIs instead of the cross-runtime n-api (which bun already implement)

We can probably look at [node-jsc](https://github.com/mceSystems/node-jsc/tree/master/deps/jscshim/src) to see how to do this.

**Comment by chriskrogh** [comment]
@Jarred-Sumner @colinhacks would it be possible for @faire to put a bounty on this? `node-canvas` is the last holdout for us adopting `bun install` in our frontend monorepo. bun would save us a lot of time in local dev & CI. https://github.com/oven-sh/bun/issues/5835#issuecomment-1943009418

(some of our testing deps link `node_canvas`: `mocha`, `jsdom`, `jest-environment-jsdom`)

**Comment by 7f8ddd** [comment]
> @Jarred-Sumner @colinhacks would it be possible for @Faire to put a bounty on this? `node-canvas` is the last holdout for us adopting `bun install` in our frontend monorepo. bun would save us a lot of time in local dev & CI. [#5835 (comment)](https://github.com/oven-sh/bun/issues/5835#issuecomment-1943009418)

node-canvas already has an NAPI port, you just need to build the latest from the git.
Alternatively, you can use https://github.com/Brooooooklyn/canvas, which is faster and has basically the same API.

**Comment by chriskrogh** [comment]
> node-canvas already has an NAPI port, you just need to build the latest from the git. Alternatively, you can use https://github.com/Brooooooklyn/canvas, which is faster and has basically the same API.

we don't depend on `canvas` ourselves. some of our dependencies try to [link](https://docs.npmjs.com/cli/v10/commands/npm-link) it during the installation process (`mocha`, `jsdom`, `jest-environment-jsdom`, ...) is there a way to get these dependencies to use the NAPI version?

**Comment by AdamSchinzel** [comment]
Any updates when we can expect this to land?

**Comment by gerardmarquinarubio** [comment]
> > node-canvas already has an NAPI port, you just need to build the latest from the git. Alternatively, you can use https://github.com/Brooooooklyn/canvas, which is faster and has basically the same API.
> 
> we don't depend on `canvas` ourselves. some of our dependencies try to [link](https://docs.npmjs.com/cli/v10/commands/npm-link) it during the installation process (`mocha`, `jsdom`, `jest-environment-jsdom`, ...) is there a way to get these dependencies to use the NAPI version?

@chriskrogh Yes if you are willing to do some [messy postinstall scripting](https://github.com/oven-sh/bun/issues/6608#issuecomment-2008316410), would be much easier if bun supported some nested overriding but that is also not in the roadmap afaik.

**Comment by chriskrogh** [comment]
@Jarred-Sumner gentle bump here. seems like both this issue (and related ones https://github.com/oven-sh/bun/issues/5835) have been racking up reactions. any plans on adding this to bun sometime soon?

**Comment by ignSKRRRTT** [comment]
Please 🙏🏻

**Comment by 7f8ddd** [comment]
> @Jarred-Sumner gentle bump here. seems like both this issue (and related ones #5835) have been racking up reactions. any plans on adding this to bun sometime soon?

node-canvas literally works fine. instead of telling Bun to update, tell those mfs to publish a release, because they haven't done that in forever.
